electrum/submarine_swaps.py

```diff
@@ -193,38 +211,62 @@ def __init__(self, *, wallet: 'Abstract_Wallet', lnworker: 'LNWallet'):
         for k, swap in self.swaps.items():
             if swap.prepay_hash is not None:
                 self.prepayments[swap.prepay_hash] = bytes.fromhex(k)
-        # api url
-        self.api_url = wallet.config.SWAPSERVER_URL
-        # init default min & max
-        self.init_min_max_values()
+        self.is_server = self.config.get('enable_plugin_swapserver', False)
+        self.is_initialized = asyncio.Event()
+
+    @property
+    def use_nostr(self):
+        return bool(self.config.NOSTR_RELAYS)
```

Comment:
I don't like this. `use_nostr` is now defined implicitly as an empty set of nostr relays.

in `create_sm_transport` above the user is asked whether to enable nostr, but this is conditional on `use_nostr` and the existence of a selected npub. if no npub is found or selected, the user is presented with this question again later. 

Better to make this a separate bool config option, and expose in the preferences. We then only have to ask this once.

Reply:
The goal is not to let the user choose between `http` and `nostr`. `http` is kept only for testing. 
So there is no need to expose another setting.

Reply:
ok, I changed this slightly, now the client-side choice between http and nostr depends on `SWAPSERVER_URL` being set. That way, we don't have to nuke the relay set in order to enable http.